src/common/asb/Asb.c

```diff
@@ -2665,120 +2670,106 @@ static int RemediateEnsureNoexecNosuidOptionsEnabledForAllNfsMounts(char* value,
 static int RemediateEnsureAllTelnetdPackagesUninstalled(char* value, void* log)
 {
     UNUSED(value);
-    UNUSED(log);
-    return 0; //TODO: add remediation respecting all existing patterns
+    return UninstallPackage(g_allTelnetd, log);
 }
 
 static int RemediateEnsureAllEtcPasswdGroupsExistInEtcGroup(char* value, void* log)
 {
     UNUSED(value);
-    UNUSED(log);
-    return 0; //TODO: add remediation respecting all existing patterns
+    return SetAllEtcPasswdGroupsToExistInEtcGroup(log);
 }
 
 static int RemediateEnsureNoDuplicateUidsExist(char* value, void* log)
 {
     UNUSED(value);
-    UNUSED(log);
-    return 0; //TODO: add remediation respecting all existing patterns
+    return SetNoDuplicateUids(log);
 }
 
 static int RemediateEnsureNoDuplicateGidsExist(char* value, void* log)
 {
     UNUSED(value);
-    UNUSED(log);
-    return 0; //TODO: add remediation respecting all existing patterns
+    return SetNoDuplicateGids(log);
 }
 
 static int RemediateEnsureNoDuplicateUserNamesExist(char* value, void* log)
 {
     UNUSED(value);
-    UNUSED(log);
-    return 0; //TODO: add remediation respecting all existing patterns
+    return SetNoDuplicateUserNames(log);
 }
 
 static int RemediateEnsureNoDuplicateGroupsExist(char* value, void* log)
 {
     UNUSED(value);
-    UNUSED(log);
-    return 0; //TODO: add remediation respecting all existing patterns
+    return SetNoDuplicateGroupNames(log);
 }
 
 static int RemediateEnsureShadowGroupIsEmpty(char* value, void* log)
 {
     UNUSED(value);
-    UNUSED(log);
-    return 0; //TODO: add remediation respecting all existing patterns
+    return SetShadowGroupEmpty(log);
 }
 
 static int RemediateEnsureRootGroupExists(char* value, void* log)
 {
     UNUSED(value);
-    UNUSED(log);
-    return 0; //TODO: add remediation respecting all existing patterns
+    return RepairRootGroup(log);
 }
 
 static int RemediateEnsureAllAccountsHavePasswords(char* value, void* log)
 {
     UNUSED(value);
-    UNUSED(log);
-    return 0; //TODO: add remediation respecting all existing patterns
+    // We cannot automatically add passwords for user accounts that can login and do not have passwords set.
+    // If we try for example to run a command such as usermod, the command line can reveal that password 
+    // in clear before it gets encrypted and saved. Thus we simply delete such accounts:
+    return RemoveUsersWithoutPasswords(log);
```

Comment:
Is this behavior worthy of discussion? This seems like one possible remediation (perhaps the right one)... I assume we are defining the remediation behavior here (because it was previously not possible). Can the users either be disabled or expired?